WordPress/gutenberg
# Components: refactor `Flex` to pass `exhaustive-deps`

## What?
Updates the `Flex` component to satisfy the `exhaustive-deps` eslint rule.

## Why?
Part of the effort in https://github.com/WordPress/gutenberg/pull/41166 to apply `exhuastive-deps` to the Components package

## How?
Including `rtl.watch()` in the dep array can't be checked by ESLint, so we need to extract it into its own variable that _can_ be checked.

Even with that done, we need to disable `exhaustive-deps` here for now, because we're relying on a value from outside of the `useEffect` to trigger it, making the dependency invalid. See https://github.com/WordPress/gutenberg/pull/41254#pullrequestreview-982176724 for more context. 

## Testing Instructions
1. From your local Gutenberg directory, run `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/flex`
2. Confirm that the linter returns no errors
3. Run Storybook locally, confirm the components stories and/or docs still work as expected
